framework/src/variables/MooseVariableFV.C

```diff
@@ -1034,6 +1066,65 @@ MooseVariableFV<OutputType>::clearAllDofIndices()
   _neighbor_data->clearDofIndices();
 }
 
+template <typename OutputType>
+ADReal
+MooseVariableFV<OutputType>::operator()(const MooseVariableFV<OutputType>::FaceArg & face) const
+{
+  const FaceInfo * const fi = std::get<0>(face);
+  mooseAssert(fi, "The face information must be non-null");
+  if (isExtrapolatedBoundaryFace(*fi))
+    return getExtrapolatedBoundaryFaceValue(*fi);
+  else if (isInternalFace(*fi))
+    return getInternalFaceValue(face);
+  else
+  {
+    mooseAssert(isDirichletBoundaryFace(*fi), "We've run out of face types");
+    return getDirichletBoundaryFaceValue(*fi);
+  }
+}
+
+template <typename OutputType>
+ADReal
+MooseVariableFV<OutputType>::operator()(const ElemAndFaceArg & elem_and_face) const
+{
+  const Elem * const requested_elem = std::get<0>(elem_and_face);
+  mooseAssert(requested_elem != remote_elem,
+              "If the requested element is remote then I think we've messed up our ghosting");
+
+  if (requested_elem && this->hasBlocks(requested_elem->subdomain_id()))
+    return getElemValue(requested_elem);
+  else
+  {
+    const FaceInfo * const fi = std::get<1>(elem_and_face);
+    mooseAssert(fi, "We need a FaceInfo");
+    mooseAssert((requested_elem == &fi->elem()) || (requested_elem == fi->neighborPtr()),
+                "The requested element should match something from the FaceInfo");
+    const Elem * const elem_across =
+        (requested_elem == &fi->elem()) ? fi->neighborPtr() : &fi->elem();
+    return getNeighborValue(requested_elem, *fi, getElemValue(elem_across));
+  }
+}
+
+template <typename OutputType>
+ADReal
+MooseVariableFV<OutputType>::operator()(
+    const std::pair<Moose::ElementType, unsigned int> & tqp) const
+{
+  const auto elem_type = tqp.first;
+  const auto qp = tqp.second;
+  switch (elem_type)
+  {
+    case Moose::ElementType::Element:
+      return adSln()[qp];
+
+    case Moose::ElementType::Neighbor:
+      return adSlnNeighbor()[qp];
+
+    default:
+      mooseError("Unrecognized element type");
+  }
+}
```

Comment:
This still assumes that we have the variable currently init'd with the correct element.  It would be nice if we could eliminate this assumption and just make the element a part of the calling args somehow.  Of course I'm sure that's exactly as easy as it sounds.

Reply:
As per our discussion yesterday - I think including the quadrature rule here is a good approach - along with something like element/local hilbert indices to uniquely identify the qps for caching.

Reply:
I added a quadrature rule to the `QpArg` overload but I've left the other "quadrature" overload that takes `ElementType` without it since that argument doesn't contain an element either (and hence a quadrature rule would be useless). The former represents the on-the-fly paradigm while the latter is in-line with old-school MOOSE that relies on variable pre-initialization. I could be persuaded to remove this latter overload.